agent/command/results_gotest.go

```diff
@@ -25,7 +26,8 @@ type goTestResults struct {
 
 	// Optional, when set to true, causes this command to be skipped over without an error when
 	// no files are found to be parsed.
-	OptionalOutput bool `mapstructure:"optional_output" plugin:"expand"`
+	OptionalOutput     string `mapstructure:"optional_output" plugin:"expand"`
+	OptionalOutputBool bool
```

Comment:
nit: since we don't want to read this in from mapstructure at all, and we just want to use it internally, you can just keep this private (lowercase optionalOutputBool)

Reply:
Also, rather than calling this the type, can we call it OutputIsOptional or something, for readability?

Reply:
done